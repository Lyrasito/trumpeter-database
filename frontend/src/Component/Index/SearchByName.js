import React from "react";
import { ReactComponent as SearchLogo } from "../../svg/Search.svg";

class SearchByName extends React.Component {
  state = {
    name: "",
  };

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  };

  search = () => {
    this.props.searchByName(this.state.name);
    this.setState({ name: "" });
    this.props.haveSearched();
  };

  render() {
    return (
      <div className="nameSearch">
        <input
          id="nameSearch"
          name="name"
          onChange={this.handleChange}
          value={this.state.name}
        ></input>
        <br />

        <button className="submit" onClick={this.search}>
          Search
          <SearchLogo />
        </button>

        <br />
      </div>
    );
  }
}

export default SearchByName;
