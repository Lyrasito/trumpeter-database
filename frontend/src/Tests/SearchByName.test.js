import SearchByName from "../Component/Index/SearchByName";
import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

describe("<SearchByName />", () => {
  it("renders properly", () => {
    const wrapper = shallow(<SearchByName />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
