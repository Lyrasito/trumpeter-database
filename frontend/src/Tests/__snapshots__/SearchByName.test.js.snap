// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchByName /> renders properly 1`] = `
<div
  className="nameSearch"
>
  <input
    id="nameSearch"
    name="name"
    onChange={[Function]}
    value=""
  />
  <br />
  <button
    className="submit"
    onClick={[Function]}
  >
    Search
    <ForwardRef(SvgSearch) />
  </button>
  <br />
</div>
`;
